src/dispatch_glx.c

```diff
@@ -109,3 +111,27 @@ epoxy_has_glx_extension(Display *dpy, int screen, const char *ext)
      */
     return epoxy_extension_in_string(glXQueryExtensionsString(dpy, screen), ext);
 }
+
+PUBLIC bool
+epoxy_has_glx(Display *dpy)
+{
+#if !PLATFORM_HAS_GLX
+    return false;
+#else
+    /* If the application hasn't explicitly called some of our GLX
+     * code but already figured out how to load GL symbols, then we
+     * simply inspect the address space.
+     *
+     * We explicitly don't use the epoxy wrappers, here, to avoid
+     * unintended side effects within our own layer of indirection.
+     */
+    Bool (* pf_glXQueryExtension) (Display *, int *, int *);
+    int error_base, event_base;
+
+    pf_glXQueryExtension = epoxy_glx_dlsym("glXQueryExtensions");
```

Comment:
Don't you want to be calling a variant of `epoxy_glx_dlsym` that passes `false` for the `exit_on_fail` parameter?

I think this implementation would just exit 1 in an environment with no libGL.so.1, where we want it to return false.